### PR TITLE
Worldmap scripting

### PR DESCRIPTION
**Pull request by Semphriss**
This PR gives access to the `settings` object in worldmaps, thus permitting usage of ambient lighting and music from scripting.

### CODE REVIEW

File: src/scripting/worldmap.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+namespace scripting {
+
+WorldMap::WorldMap(::worldmap::WorldMap* parent) :
+  GameObjectManager(parent),
+  m_parent(parent)
+{
+}
+
+float
+WorldMap::get_tux_x()
```
Comment (tobbi) [inline src/scripting/worldmap.cpp:30]: const

File: src/scripting/worldmap.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+{
+  return m_parent->get_tux_pos().x;
+}
+
+float
+WorldMap::get_tux_y()
```
Comment (tobbi) [inline src/scripting/worldmap.cpp:36]: const

File: src/scripting/worldmap.hpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+public:
+  WorldMap(::worldmap::WorldMap* parent);
+
+private:
+  WorldMap(const WorldMap&) = delete;
+  WorldMap& operator=(const WorldMap&) = delete;
+#endif
+
+public:
+  float get_tux_x();
```
Comment (tobbi) [inline src/scripting/worldmap.hpp:46]: mark const here as well.

File: src/worldmap/worldmap.cpp [hunk 4]
```diff
@@ -712,6 +715,12 @@ WorldMap::set_passive_message(const std::string& message, float time)
    m_passive_message_timer.start(time);
 }
 
+Vector
+WorldMap::get_tux_pos()
+{
```
Comment (tobbi) [inline src/worldmap/worldmap.cpp:720]: const

File: src/worldmap/worldmap.hpp [hunk 5]
```diff
@@ -158,6 +158,8 @@ class WorldMap final : public GameObjectManager,
 
   Camera& get_camera() const { return *m_camera; }
 
+  Vector get_tux_pos();
```
Comment (tobbi) [inline src/worldmap/worldmap.hpp:161]: ...and const

**Review by tobbi — CHANGES_REQUESTED** [review]

**Review by tobbi — APPROVED** [review]

**Comment by tobbi** [comment]
Please rebase.

**Comment by Semphriss** [comment]
Superseded by #1692, which got properly rebased

**Pull request closed without merging**